docs/frontend_architecture/unit_testing.rst

```diff
@@ -21,4 +21,79 @@ Tests are written in JavaScript, and placed in the 'assets/test' folder. An exam
   });
 
 
-Vue.js components can also be tested. The management plugin contains an example (*kolibri/plugins/management/assets/test/management.js*) where the component is bound to a temporary DOM node, changes are made to the state, and assertions are made about the new component structure.
+For the frontend testing of ``Vue.js`` components, we make use of `Vue Testing Library <https://testing-library.com/docs/vue-testing-library/intro/>`__
+and the associated ecosystem.
+
+It is based on the philosphy that “The more your tests resemble the way your software is used, the more confidence they can give you." Rather than dealing with instances of rendered Vue components, it allows our tests to work with actual DOM nodes in the same way the user would. This deals to more user centric and a better quality of tests generally, and thus we have been making efforts to use the same (in constrast to the earlier used `Vue Test Utils <https://v1.test-utils.vuejs.org/>`__).
+
+If you have never worked with Vue Testing Library's (VTL) before, it is highly recommeed to the `VTL examples page <https://testing-library.com/docs/vue-testing-library/examples>`__ to see the library in action and understand it's use. If you're comfortable with the same, you can either:
+
+-  Read some of the exisiting tests in Kolibri to see how we make use of the same.
+-  Checkout the `test template <TODO>`__ and the `style guide <TODO>`__ to start writing your own!
+
+Style Guide
+-----------
+
+Here are some of the conventions that we prefer to follow while testing so that uniformity is maintained across the codebase:
+
+Naming Conventions
+~~~~~~~~~~~~~~~~~~
+
+Each folder is expected to have a ``__tests__`` folder, which would contain all the test files for the files in the same (whether ``.js`` or ``.vue``). The test files are expected to follow the naming convention ``<Name of the file being tested>.spec.js``.
+
+Use of ``renderComponent`` function
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+All the files testing ``Vue.js`` components are expected to have their first function as ``renderComponent``. The job of this function should be to setup all the necesary mocks to render the component in the testing environment properly. It can also revieve an optional argument ``props``, which can be used to overwrite:
+
+-  The default props passed to the component being rendered
+-  Configuration passed to other mocks/stubs (like the values the getters for store mock should return, arguments to Vue Router etc)
+
+Queries
+~~~~~~~
+
+VTL provides a number of `queries <https://testing-library.com/docs/vue-testing-library/cheatsheet#queries>`__ that can be used to query the DOM nodes. There are primarily three types of queries: ``get``, ``query`` and ``find``. All three of them have two variants: one to query just a single DOM node, and the other to query multiple. Please ensure that you are using the correct kind of the query according to the particular usecase.
```

Comment:
Not sure about rewriting this in our docs, I think we can refer to the testing library docs and say that we follow their recommended priority. What are your thoughts on this? @MisRob

Reply:
I wouldn't mind having it briefly mentioned as a reminder since it's important, so perhaps something along the lines "Pay attention to the recommended priority of using queries: `get`, `query`, and `find`. See <link> for more." would balance both perspectives?

Reply:
Got it!